src/incremental.jl

```diff
@@ -0,0 +1,183 @@
+    @eval(Sys, STDLIB = joinpath(\$_bindir, "..", "share", "julia", "stdlib", string('v', (VERSION.major), '.', VERSION.minor)))
+    """
+end
+
+function Include(path)
+  """
+  M = Module()
+  # Include into anonymous module to not polute namespace
+  @eval(M, (Base.include(\$M, $(repr(path)))))
+  """
+end
+
+"""
+The command to pass to julia --output-o, that runs the julia code in `path` during compilation.
+"""
+function PrecompileCommand(path)
+    InitBase() * Fix30479() * InitREPL() * Include(path)
+end
+
+
+"""
+    extract_flag(flag, jl_cmd = Base.julia_cmd())
+
+Extracts the value for `flag` from the current julia cmd
+"""
+function extract_flag(flag, jl_cmd = Base.julia_cmd())
+    for elem in jl_cmd.exec
+        if startswith(elem, flag)
+            return elem
+        end
+    end
+    return nothing
+end
+
+function extract_flag_value(flag, jl_cmd = Base.julia_cmd())
+    val = extract_flag(flag, jl_cmd)
+    replace(val, flag => "")
+end
+current_systemimage() = extract_flag_value("-J")
+
+"""
+    add_command!(cmd, ignore, jl_cmd, flag, value)
+
+Adds a command to `cmd` - either choosing the julia default for value == nothing,
+or adds value!. If ignore = true, and command not in jl_cmd && value == nothing,
+command gets not added.
+"""
+function add_command!(cmd, ignore, jl_cmd, flag, value)
+    new_cmd = if value == nothing
+        extract_flag(flag, jl_cmd)
+    else
+        short = !startswith(flag, "--")
+        string(flag, short ? "" : "=", value)
+    end
+    if new_cmd == nothing
+        ignore || @warn("Flag $flag not present in julia-cmd, but is set to nothing - can't add the flag")
+    else
+        push!(cmd, new_cmd)
+    end
+end
+
+"""
+    run_julia(
+        code::String;
+        g = nothing, O = nothing, output_o = nothing, J = nothing,
+        startup_file = "no"
+    )
+
+Runs `code` in a new julia command!
+You can overwrite any julia command line flag by setting it to a value.
+If nothing is chosen, it will default to the value of the current julia process.
+"""
+function run_julia(code::String; kw...)
+    run(julia_code_cmd(code; kw...))
+end
+
+function julia_code_cmd(
+        code::String;
+        g = nothing, O = nothing, output_o = nothing, J = nothing,
+        startup_file = nothing, trace_compile = nothing, compile = nothing
+    )
+
+    jl_cmd = Base.julia_cmd()
+    cmd = `$(jl_cmd.exec[1])` # extract julia exe
+
+    add_command!(cmd.exec, false, jl_cmd, "-O", O)
+    add_command!(cmd.exec, false, jl_cmd, "-J", J)
+
+    add_command!(cmd.exec, true, jl_cmd, "-g", g)
+    add_command!(cmd.exec, true, jl_cmd, "--output-o", output_o)
+    add_command!(cmd.exec, true, jl_cmd, "--startup-file", startup_file)
+    add_command!(cmd.exec, true, jl_cmd, "--compile", compile)
+    add_command!(cmd.exec, true, jl_cmd, "--trace-compile", trace_compile)
+    # for better debug, let's not make a tmp file and immediately delete it!
+    file = sysimg_folder("run_julia_code.jl")
```

Comment:
Is this created somewhere?

Reply:
The folder? Should be, it's the folder I store stuff related to system image creation.